### PR TITLE
Various Small Fixes

### PR DESCRIPTION
**Pull request by tsortwehttam**
OK to merge.

Short-ish review.

Asana task link: n/a, various

Changes in this PR: Please see the commit messages

Completed checkin tasks:

- [x] Did manual testing of interrelated functionality
- [x] Ran the automated tests and linted the code

### CODE REVIEW

File: packages/@haiku/core/src/HaikuComponent.ts [hunk 1]
```diff
@@ -31,13 +31,20 @@ const HAIKU_ID_ATTRIBUTE = 'haiku-id';
...
-    throw new Error('Bytecode must define template');
+  if (!bytecode.timelines) {
+    console.warn('Adding missing timeline object to bytecode');
+    bytecode.timelines = {};
+  }
+
+  if (!bytecode.timelines[DEFAULT_TIMELINE_NAME]) {
+    console.warn('Adding missing default timeline to bytecode');
+    bytecode.timelines[DEFAULT_TIMELINE_NAME] = {};
   }
```
Comment (stristr) [inline packages/@haiku/core/src/HaikuComponent.ts:48]: :+1:

File: packages/haiku-bytecode/src/TimelineProperty.js [hunk 2]
```diff
@@ -163,6 +163,7 @@ function addPropertyGroup (timelinesObject, timelineName, componentId, elementNa
  */
 function getPropertySegmentsBase (timelinesObject, timelineName, componentId, outputName) {
   const selector = getSelectorForComponentId(componentId)
+  if (!timelinesObject) return null
   if (!timelinesObject[timelineName]) return null
   if (!timelinesObject[timelineName][selector]) return null
   return timelinesObject[timelineName][selector][outputName]
```
Comment (stristr) [inline packages/haiku-bytecode/src/TimelineProperty.js:169]: Nit:

`if (!timelinesObject || !timelinesObject[timelineName] || !timelinesObject[timelineName][selector]) return null`

File: packages/haiku-bytecode/src/TimelineProperty.js [hunk 3]
```diff
@@ -175,6 +176,7 @@ function getPropertySegmentsBase (timelinesObject, timelineName, componentId, ou
  */
 function getPropertiesBase (timelinesObject, timelineName, componentId) {
   const selector = getSelectorForComponentId(componentId)
+  if (!timelinesObject) return null
   if (!timelinesObject[timelineName]) return null
```
Comment (stristr) [inline packages/haiku-bytecode/src/TimelineProperty.js:180]: Nit:

`if (!timelinesObject || !timelinesObject[timelineName]) return null`

File: packages/haiku-plumbing/src/Master.js [hunk 4]
```diff
@@ -334,16 +331,10 @@ export default class Master extends EventEmitter {
...
-              // Since module change results in a (heavy) stage reload, we want to do it
-              // only if we really need to, i.e. if the file contents have changed
-              if (file.previous !== file.contents) {
-                this._mod.handleModuleChange(file)
-              }
-            } else {
-              this._filesLoadedOnce[file.relpath] = true
+            // Since module change results in a (heavy) stage reload, we want to do it
+            // only if we really need to, i.e. if the file contents have changed
+            if (file.previous !== file.contents) {
```
Comment (stristr) [inline packages/haiku-plumbing/src/Master.js:336]: qq, this is a string comparison, right?

File: packages/haiku-plumbing/src/MasterModuleProject.js [hunk 5]
```diff
@@ -37,10 +37,12 @@ export default class MasterModuleProject extends EventEmitter {
   }
 
   maybeSendComponentReloadRequest (file) {
+    const lastWrite = (parseInt(file.dtLastWriteStart) || 0) + LAST_WRITE_MARGIN_OF_ERROR_MILLISECONDS
```
Comment (stristr) [inline packages/haiku-plumbing/src/MasterModuleProject.js:40]: Fewer potential effects: `parseInt(file.dtLastWriteStart || 0)`

File: packages/haiku-plumbing/src/Plumbing.js [hunk 6]
```diff
@@ -460,7 +466,15 @@ export default class Plumbing extends StateObject {
...
+        return websocket.send(data, (err) => {
+          if (err) {
+            logger.error(`[plumbing] websocket client request error`, err)
+            // In case the websocket disconnects, just remove the client instead of crashing
+            if (err.message.match(/not opened/) || err.message.match(/EPIPE/)) {
+              this.removeWebsocketClient(websocket)
+            }
+          }
+        })
       } else {
```
Comment (stristr) [inline packages/haiku-plumbing/src/Plumbing.js:478]: Curious if this lets us upgrade `ws` package. Not critical now.

**Review by stristr — COMMENTED** [review]
Nits but LGTM.

File: packages/haiku-plumbing/src/Master.js (hunk 4 above)
Comment (stristr) [reply packages/haiku-plumbing/src/Master.js:336]: (asking because there's a potentially significant perf improvement to be had here [table for now])

Comment (tsortwehttam) [reply packages/haiku-plumbing/src/Master.js:336]: Yes, string comparison.

**Comment by tsortwehttam** [comment]
Going to merge, but as usual we should QA before releasing to the wild.

**Pull request merged**